File: auction-app-backend/src/main/java/com/internship/auctionapp/services/NotificationService.java
package com.internship.auctionapp.services;

import com.internship.auctionapp.models.Notification;
import com.internship.auctionapp.requests.CreateNotificationRequest;

import java.util.List;
import java.util.UUID;

public interface NotificationService {
    List<Notification> getAllNotifications();

    Notification createNotification(CreateNotificationRequest createNotificationRequest);

    Notification getNotifications(UUID userId, UUID productId);
}

File: auction-app-ui/src/services/notificationService.ts
import { Notification } from 'models/notification';

import agent from 'lib/agent';

const BASE_URL = '/notifications';

const notificationService = {
  getLatestNotification: (userId: string, productId: string) =>
    agent.get<Notification[]>(
      `${BASE_URL}/search?userId=${userId}&productId=${productId}`
    ),
};

export default notificationService;
